api/src/pcapi/local_providers/allocine/allocine_stocks.py

```diff
@@ -209,9 +216,30 @@ def fill_stock_attributes(self, allocine_stock: Stock):  # type: ignore [no-unty
             allocine_stock.quantity = self.quantity
 
         if "price" not in allocine_stock.fieldsUpdated:
-            allocine_stock.price = self.apply_allocine_price_rule(allocine_stock)
-
-    def apply_allocine_price_rule(self, allocine_stock: Stock) -> decimal.Decimal:
+            price = self.apply_allocine_price_rule(allocine_stock)
+            allocine_stock.price = price
+            if not allocine_stock.priceCategory:
+                if self.priceCategoryByOffer and allocine_stock.offerId in self.priceCategoryByOffer:
+                    allocine_stock.priceCategory = self.priceCategoryByOffer[allocine_stock.offerId]
+                else:
+                    price_category = self.get_or_create_allocine_price_category(price, allocine_stock.offerId)
+                    allocine_stock.priceCategory = price_category
+                    self.priceCategoryByOffer[allocine_stock.offerId] = price_category
```

Comment:
(style) tu n'as pas besoin du `if self.priceCategoryByOffer`, grâce à ce qu'il y a à droite du `and`, ni du `else` parce que les deux branches finissent sur la même instruction

```suggestion
                if not allocine_stock.offerId in self.priceCategoryByOffer:
                    self.priceCategoryByOffer[allocine_stock.offerId] = self.get_or_create_allocine_price_category(price, allocine_stock.offerId)
                allocine_stock.priceCategory =  self.priceCategoryByOffer[allocine_stock.offerId]
```